projects/Buttons_and_ADC/src/main.c

```diff
@@ -0,0 +1,64 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+static void prv_button_interrupt_handler(const GpioAddress *adc_address, void *context) {
+  GpioAddress *adc_position = context;
+  uint16_t adc = 0;
+  adc_read_converted_pin(*adc_position, &adc);
+  LOG_DEBUG("ADC reading is: %d\n", adc);
+}
+
+int main(void) {
+  interrupt_init();
+  gpio_init();
+  gpio_it_init();
+
+  // Button address
+  GpioAddress button_addresses = {
+    .port = GPIO_PORT_B,
+    .pin = 2,
+  };
+
+  // Button setting
+  GpioSettings button_setting = {
+    .direction = GPIO_DIR_IN,
+    .state = GPIO_STATE_LOW,
+    .alt_function = GPIO_ALTFN_NONE,
+    .resistor = GPIO_RES_NONE,
+  };
+
+  // Adc address
+  GpioAddress adc_addresses = {
+    .port = GPIO_PORT_A,
+    .pin = 6,
+  };
+
+  // Adc setting
+  GpioSettings adc_setting = {
+    .direction = GPIO_DIR_IN,
+    .state = GPIO_STATE_LOW,
+    .alt_function = GPIO_ALTFN_ANALOG,
+    .resistor = GPIO_RES_NONE,
+  };
+
+  static InterruptSettings s_interrupt_settings = {
+    .type = INTERRUPT_TYPE_INTERRUPT,
+    .priority = INTERRUPT_PRIORITY_NORMAL,
+  };
+
+  gpio_init_pin(&adc_addresses, &adc_setting);
```

Comment:
The GPIO pin for the button must be initialized as well. It looks like you created `button_settings` but never actually used it.